src/test/java/com/github/thorbenkuck/netcom2/network/server/RemoteObjectRegistrationImplTest.java

```diff
@@ -1,42 +1,355 @@
+		String methodName = "aMethod";
+		Object[] parameters = null;
+		RemoteAccessCommunicationRequest communicationRequest = new RemoteAccessCommunicationRequest(methodName, classToAssign, uuid, parameters);
+
+		//Act
+		remoteObjectRegistration.register(remoteObject, classToAssign);
+		remoteObjectRegistration.run(communicationRequest);
+
+		//Assert
+		verify(remoteObject, never()).aMethod();
 	}
 
 	@Test
-	public void register1() throws Exception {
+	public void registerOverrideProhibited() throws Exception {
+		//Arrange
+		RemoteObjectRegistrationImpl remoteObjectRegistration = new RemoteObjectRegistrationImpl();
+		TestRemoteObject remoteObject = spy(new TestRemoteObject());
+		Class<?> classToAssign = TestRegistrationOverrideProhibitedRemoteObject.class;
+		String methodName = "aMethod";
+		UUID uuid = UUID.fromString(UUID_SEED_1);
+		Object[] parameters = null;
+		RemoteAccessCommunicationRequest communicationRequest = new RemoteAccessCommunicationRequest(methodName, classToAssign, uuid, parameters);
+
+		//Act
+		remoteObjectRegistration.register(remoteObject, classToAssign);
+		remoteObjectRegistration.run(communicationRequest);
+
+		//Assert
+		verify(remoteObject, never()).aMethod();
+	}
+
+	@Test
+	public void registerSuccess() throws Exception {
+		//Arrange
+		RemoteObjectRegistrationImpl remoteObjectRegistration = new RemoteObjectRegistrationImpl();
+		TestRemoteObject remoteObject = spy(new TestRemoteObject());
+		Class<?> classToAssign = TestRemoteObject.class;
+		String methodName = "aMethod";
+		UUID uuid = UUID.fromString(UUID_SEED_1);
+		Object[] parameters = null;
+		RemoteAccessCommunicationRequest communicationRequest = new RemoteAccessCommunicationRequest(methodName, classToAssign, uuid, parameters);
+
+		//Act
+		remoteObjectRegistration.register(remoteObject, classToAssign);
+		remoteObjectRegistration.run(communicationRequest);
+
+		//Assert
+		verify(remoteObject).aMethod();
 	}
 
 	@Test
-	public void hook() throws Exception {
+	public void hookInterface() throws Exception {
+		//Arrange
+		RemoteObjectRegistrationImpl remoteObjectRegistration = new RemoteObjectRegistrationImpl();
+		TestRemoteObjectWithInterface object = spy(new TestRemoteObjectWithInterface());
+		String methodName = "anInterfaceMethod";
+		Class<TestRemoteObjectWithInterface> clazz = TestRemoteObjectWithInterface.class;
+		Object[] parameters = null;
+		UUID uuid = UUID.fromString(UUID_SEED_1);
+		RemoteAccessCommunicationRequest request = new RemoteAccessCommunicationRequest(methodName, clazz, uuid, parameters);
+
+		//Act
+		remoteObjectRegistration.hook(object);
+		remoteObjectRegistration.run(request);
+
+		//Assert
+		verify(object).anInterfaceMethod();
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void hookNull() throws Exception {
+		//Arrange
+		RemoteObjectRegistrationImpl remoteObjectRegistration = new RemoteObjectRegistrationImpl();
+
+		//Act
+		remoteObjectRegistration.hook(null);
+
+		//Assert
```

Comment:
If the Exception is the required criteria for this test to succeed, a fail() at assert should be inserted.